docs/User/HowTo/faq.md

```diff
@@ -7,64 +7,88 @@ This document provides additional examples.
-transformed in multiple ways using monadic functions. Then you want to zip the
-results together and consume the resulting stream concurrently.
+Transform a stream in multiple ways, generating multiple transformed
+streams and then zip the corresponding elements of each resulting stream
+together to create a single transformed stream.
 
-First, distribute an item to multiple effectful consumer concurrently, for
-example:
+Distributing a value to a stream of consumers concurrently:
+
+```haskell ghci
+{-# LANGUAGE FlexibleContexts #-}
 
-```haskell
 import Data.Function ((&))
-import qualified Streamly.Prelude as Stream
+import qualified Streamly.Data.Fold as Fold
+import qualified Streamly.Data.Stream.Prelude as Stream
 
 f1 x =
-      Prelude.map ($ x) [return . (+ 1), return . (+ 2)]
-    & Stream.fromListM
-    & Stream.fromAhead
-    & Stream.toList
+    Stream.fromList [return . (+ 1), return . (+ 2)] -- Stream of functions
+        & fmap ($ x)                                 -- Stream of lazy actions
+        & Stream.parSequence (Stream.ordered True)   -- Evaluate concurrently
+        & Stream.fold Fold.toList                    -- Fold to list
 ```
 
-Alternatively, if you want to concurrently zip different types of outputs, you
-can use `Applicative` composition:
+Use `parApply` to zip streams concurrently. Here, we zip three singleton
+streams:
+
+```haskell ghci
 
-```haskell
-import Data.Maybe (fromJust)
 f2 x =
-    (,,)
-      <$> Stream.fromEffect (return $ show x)
-      <*> Stream.fromEffect (return $ x + 1)
-      <*> Stream.fromEffect (return $ fromIntegral x / 2)
-  & Stream.fromAsync
-  & Stream.head
-  & fmap fromJust
+  let app = Stream.parApply id
+  in (,,)
+      `fmap` Stream.fromEffect (return $ show x)
+      `app`  Stream.fromEffect (return $ x + 1)
+      `app`  Stream.fromEffect (return $ fromIntegral x / 2)
+  & Stream.fold Fold.one
 ```
 
-Then apply the function `f` concurrently to your input stream:
+Applying a function concurrently to your input stream:
 
-```haskell
+```haskell ghci
 g f xs =
   Stream.fromList xs
-    & Stream.mapM f
-    & Stream.fromAhead
-    & Stream.toList
+    & Stream.parMapM (Stream.ordered True) f
+    & Stream.fold Fold.toList
 ```
 
-```
+You can now use the concurrent map to pipe each element through multiple
+transformations using the distribute/zip operations.
+
+```haskell docspec
 >>> g f1 [1,2,3,4::Int]
 [[2,3],[3,4],[4,5],[5,6]]
 
 >>> g f2 [1,2,3,4::Int]
-[("1",2,0.5),("2",3,1.0),("3",4,1.5),("4",5,2.0)]
+[Just ("1",2,0.5),Just ("2",3,1.0),Just ("3",4,1.5),Just ("4",5,2.0)]
 ```
 
-## Sliding Window
+Instead of using `parApply` directly, you can use `mkZipType` to
+create a zip Applicative newtype so that you can use the `Applicative`
+instance.
 
+```haskell
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+import Streamly.Internal.Data.Stream.TypeGen
```

Comment:
This needs to be changed after the other PR is merged. You can merge that first, fix this and merge it.